### ISSUE TITLE
Error making: ccls-git from Archlinux/AUR.

### ISSUE THREAD

**Issue by SyrakuShaikh** [MaskRay/ccls#25]
I just tried to install `ccls` from `aur/ccls-git` and came into this error.

```make
......
[ 94%] Building CXX object CMakeFiles/ccls.dir/src/messages/workspace_didChangeConfiguration.cc.o
[ 96%] Building CXX object CMakeFiles/ccls.dir/src/messages/workspace_didChangeWatchedFiles.cc.o
[ 98%] Building CXX object CMakeFiles/ccls.dir/src/messages/workspace_symbol.cc.o
[100%] Linking CXX executable ccls
/usr/bin/ld: cannot find -lLLVM_LIBRARY-NOTFOUND
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/ccls.dir/build.make:949: ccls] Error 1
make[1]: *** [CMakeFiles/Makefile2:68: CMakeFiles/ccls.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
Error making: ccls-git
```

seems that it is related with `llvm-libs`?

here are my clang-llvm toolchains installed:

```pacman
local/clang-analyzer-svn 7.0.0svn_r336353-1 (llvm-toolchain-svn)
    Source code analysis tool for Clang, supporting C, C++, and Objective-C
local/clang-compiler-rt-svn 7.0.0svn_r336353-1 (llvm-toolchain-svn)
    The "compiler-rt" set of runtime libraries for Clang and LLVM
local/clang-svn 7.0.0svn_r336353-1 (llvm-toolchain-svn)
    C language family frontend for LLVM
local/clang-tools-extra-svn 7.0.0svn_r336353-1 (llvm-toolchain-svn)
    Standalone tools for Clang: syntax checking, formatting, refactoring, etc.
local/liblphobos 1:1.10.0-1 (dlang dlang-ldc)
    A D Compiler based on the LLVM Compiler Infrastructure including D runtime and libphobos2
local/lld-svn 7.0.0svn_r336353-1 (llvm-toolchain-svn)
    A linker from the LLVM project
local/lldb-svn 7.0.0svn_r336353-1 (llvm-toolchain-svn)
    Next generation, high-performance debugger
local/llvm-libs-svn 7.0.0svn_r336353-1 (llvm-toolchain-svn)
    The LLVM Compiler Infrastructure (runtime libraries)
local/llvm-svn 7.0.0svn_r336353-1 (llvm-toolchain-svn)
    The LLVM Compiler Infrastructure
local/openmp-svn 5.0.r336411-1
    LLVM OpenMP Runtime Library
```

so, what should I do? Thanks!

**Comment by SyrakuShaikh** [comment]
additional info may be helpful.
This is the configure information when installing `aur/ccls-git`.

```make
==> Starting pkgver()...
==> Starting build()...
-- The CXX compiler identification is GNU 8.1.1
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'Release' as none was specified.
-- Using system Clang
-- Found Clang: /usr/lib/libclang.so (found suitable version "7.0.0", minimum required is "6.0.0") 
-- Found Curses: /usr/lib/libcurses.so  
-- Found ZLIB: /usr/lib/libz.so (found version "1.2.11") 
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/makepkg/ccls-git/src/ccls/build
```

**Comment by MaskRay** [comment]
https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=ccls-git

```
cmake -H. -Bbuild -DCMAKE_INSTALL_PREFIX=/usr -DSYSTEM_CLANG=on -DUSE_SHARED_LLVM=on
```

`-DUSE_SHARED_LLVM` instructs cmake to link `/usr/lib/libLLVM.so` instead of multiple `libLLVM*.a`.

Can you remove this option locally and retry? Note, `extra/clang 6.0.0-1` is known to have a bug (see the comment at https://aur.archlinux.org/packages/ccls-git

**Comment by SyrakuShaikh** [comment]
OK, I figure it out.

Before I installed `aur/ccls-gt`, I had noticed that `extra/clang 6.0.0-1` is said to have a bug. Thus, I replaced the whole **clang-llvm** tools by using their *-svn* version.

However, what I missed is the package `aur/llvm-libs-svn` installs its libraries with a **svn** suffix! Therefore, when I built  `ccls-git`, no `/usr/lib/libLLVM.so` was found!

After I made a system link of it, no errors occur any more.

Thanks.
